janusgraph-core/src/main/java/org/janusgraph/graphdb/util/SubqueryIterator.java

```diff
@@ -88,9 +86,13 @@ public JanusGraphElement next() {
     }
 
     @Override
-    public void close() throws Exception {
+    public void close() {
         if (isTimerRunning) {
+            assert currentIds != null;
+            indexCache.put(subQuery, currentIds);
```

Comment:
Not sure: The encapsulation of all the code within an `if (isTimerRunning)` suggests that calling close() with an already stopped timer is also a valid operation. Does it make sense to restrict the cache update to the case where the timer is still running?
From my understanding, there should be no real difference since after the timer has stopped, `currentIds` should not be altered anymore. But still I would be more comfortable if the if block only encapsulates operations that are invalid if the timer is stopped.